### PR TITLE
Build new theme using Webpack 4

### PR DESCRIPTION
**Pull request by CaptainYouz**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR migrate the new theme of BO to Webpack 4.
| Type?         | improvement
| Category?     | BO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | none
| How to test?  | Everything in the new theme should work as before.

With this migration, three modes are now available:

- `npm run build`: production mode
-  `npm run build:analyze`: production mode with bundler analyzer
-  `npm run dev`: development mode for the new theme

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10911)

### CODE REVIEW

**Comment by PierreRambaud** [comment]
Is this PR ready for review @eternoendless?

**Comment by eternoendless** [comment]
I think it is. Just FYI there's an issue with live reload, I'm under the impression that exported images are missing when you use it. Also, I think we lost the source maps.

File: admin-dev/themes/new-theme/.webpack/dev.js [hunk 1]
```diff
@@ -8,28 +7,12 @@ const LiveReloadPlugin = require('webpack-livereload-plugin');
  * @param {String} hostname Development host name, sent as a parameter to webpack. Defaults to localhost
  */
 function devConfig(hostname) {
```
Comment (eternoendless) [inline admin-dev/themes/new-theme/.webpack/dev.js]: If live reload is removed, then there's no need for `hostname` anymore. @PierreRambaud you couldn't get it to run alongside the css extraction plugin?

**Review by eternoendless — COMMENTED** [review]
Don't forget to rebase, remove the previous commit containing compiled assets, build and commit them again.

File: admin-dev/themes/new-theme/.webpack/dev.js (hunk 1 above)
Comment (PierreRambaud) [reply admin-dev/themes/new-theme/.webpack/dev.js]: Nothing was working for me for with Live reload. I remove unneeded variables.

**Review by eternoendless — DISMISSED** [review]
LGTM

**Review by jolelievre — DISMISSED** [review]
Good job @PierreRambaud

**Review by matks — APPROVED** [review]

**Comment by ntiepresta** [comment]
Hello, 
after testing this Pr, everything works well as before, except for a few points as can be seen with the images below.
Before testing
normale
test with PR
should

**Comment by ntiepresta** [comment]
Before testing
config ok
test with PR
config should

**Comment by ntiepresta** [comment]
The user profile pop-up is not correctly displayed in the pages below
1)  SELL Orders( Invoies, Delivery slips);  
                Catalog(products, stocks)
2) IMPROVE  Modules( Modules selections)
                      Modules Manager( modules, alerts, updates)
                     Design(theme & logo,)
                     Positions
                     Shipping(preferences)
                      Payment(payment methods, preferences)
International(localization, geolocation)
                    Translations)

3) Shop Parameters( General, maintenance)
            Order Setting( order setting)
             Product Setting
             Customer setting(customers)
             Traffic & SEO( Seo & urls)

4) Advanced Parameters( Information)
                     Performance
                     Administtration
                     E-mail
                     import
                     Database
                     Logs.

Before testing
pop up profil ok
test with PR
profil should

Best regards,
Reddy Ntie

**Comment by matks** [comment]
@ntiepresta Thank you for testing !

Since this PR has been hard to make and hard to rebase and the issues you reported are minor, we chose to merge this PR and handle your feedbacks in another PR. I'll take care of creating the right issue 😄

**Comment by matks** [comment]
Thank you @PierreRambaud and @CaptainYouz

**Comment by matks** [comment]
QA reported issues have been listed here: https://github.com/PrestaShop/PrestaShop/issues/12333

**Comment by eternoendless** [comment]
Reverted in #10911

**Pull request merged**